src/main/scala/ch/openolitor/stammdaten/KorbHandler.scala

```diff
@@ -268,6 +268,23 @@ trait KorbHandler extends KorbStatusHandler
 
   def deleteKorb(lieferung: Lieferung, abo: Abo)(implicit personId: PersonId, session: DBSession, publisher: EventPublisher): Option[Korb] = {
     logger.debug(s"deleteKorb lieferung: $lieferung abo: $abo")
+    abo match {
+      case _: ZusatzAbo =>
+      case _ =>
+        {
+          lieferung.lieferplanungId map { lieferplanungId =>
+            stammdatenWriteRepository.getZusatzAbos(abo.id) flatMap { zusatzabo =>
+
+              stammdatenWriteRepository.getExistingZusatzaboLieferung(zusatzabo.abotypId, lieferplanungId, lieferung.datum) map { lieferung =>
```

Comment:
Can you call the return attribute of getExistingZusatzaboLieferung  'zusatzaboLieferung' or so? As there is a function-attribute 'lieferung' and this is likelihood of confusion.